exir/emit/_emitter.py

```diff
@@ -466,6 +466,15 @@ def _tensor_spec_to_evalue(
                 and spec.extra_tensor_info.location == TensorDataLocation.EXTERNAL
             ):
                 buffer_idx = self.program_state.external_constant_hash.get(hashed, -1)
+                if buffer_idx != -1:
+                    # Save the constant tag for the external tensor
+                    if constant_tag not in self.program_state.external_constant_map:
+                        # pyre-ignore Undefined attribute [16]: `Optional` has no attribute `fully_qualified_name`.
+                        self.program_state.external_constant_map[constant_tag] = {}
+                    # pyre-ignore Undefined attribute [16]: `Optional` has no attribute `fully_qualified_name`.
+                    self.program_state.external_constant_map[constant_tag][
+                        spec.extra_tensor_info.fully_qualified_name
+                    ] = buffer_idx
```

Comment:
also please assert that constant_tag is not None
can you deduplicate this code. i think there's another place in this file doing the same logic